Source/src/events/MouseEventPayload.h

```diff
@@ -1,35 +1,23 @@
 #pragma once
 
-#include <MathGeolib.h>
-
 namespace Hachiko
 {
     struct MouseEventPayload
     {
-        enum Action
+        enum class Action
         {
-            CLICK,
-            RELEASE,
+            Click,
+            Release,
         };
 
-        MouseEventPayload(Action action, const float2& coords) :
-            action(action),
-            coords(coords)
-        {
-        }
+        explicit MouseEventPayload(Action action) : action(action) {}
```

Comment:
Was it being called implicitly somewhere or u feeling fancy?

Reply:
As also stated by sonar, it's error prone to have single parametered ctors to be implicit enabled. As it enables you to pass any data type that can be converted into Action wherever you create MouseEventPayload.